CIP-0102/README.md

```diff
@@ -0,0 +1,177 @@
+
+## Specification
+
+### 500 Royalty Datum Standard
+
+The following defines the `500` Royalty NFT standard with the registered `asset_name_label` prefix value
+
+| asset_name_label            | class        | description                                                          |
+| --------------------------- | ------------ | -------------------------------------------------------------------- |
+| 500                         | NFT          | Royalty NFT stored in a UTxO containing a datum with royalty information |
+
+#### Class
+
+The `royalty NFT` is an NFT (non-fungible token).
+
+#### Pattern
+
+The `royalty NFT` **must** have an identical `policy id` as the collection.
+
+The `asset name` **must** be `001f4d70526f79616c7479` (hex encoded), it contains the [CIP-0067][] label `500` followed by the word "Royalty".
+
+Example:\
+`royalty NFT`: `(500)Royalty`\
+`reference NFT`: `(100)Test123`
+
+#### 500 Datum Metadata
+
+The royalty info datum is specified as follows (CDDL):
+
+```cddl
+big_int = int / big_uint / big_nint
+big_uint = #6.2(bounded_bytes)
+big_nint = #6.3(bounded_bytes)
+
+optional_big_int = #6.121([big_int]) / #6.122([])
+
+royalty_recipient = #6.121([
+              address,                    ; definition can be derived from: 
+                                          ; https://github.com/input-output-hk/plutus/blob/master/plutus-ledger-api/src/PlutusLedgerApi/V1/Address.hs#L31
+              int,                        ; variable fee ( calculation: ⌊1 / (fee / 10)⌋ ); integer division with precision 10
+              optional_big_int,           ; min fee (absolute value in lovelace)
+              optional_big_int,           ; max fee (absolute value in lovelace)
+            ])
+
+royalty_recipients = [ * royalty_recipient ]
+
+; version is of type int, we start with version 1
+version = 1  
+
+; Custom user defined plutus data.
+; Setting data is optional, but the field is required
+; and needs to be at least Unit/Void: #6.121([])
+extra = plutus_data
+
+royalty_info = #6.121([royalty_recipients, version, extra])
+```
+
+#### Example of onchain variable fee calculation:
+```
+; Given a royalty fee of 1.6% (0.016)
+
+; To store this in the royalty datum
+1 / (0.016 / 10) => 625
+
+; To read it back
+10 / 625 => 0.016
+```
+Because the computational complexity of Plutus primitives scales with size, this approach significantly minimizes resource consumption.
+
+To prevent abuse, it is **recommended** that the `royalty NFT` is stored at the script address of a validator that ensures the specified fees are not arbitrarily changed, such as an always-fails validator.
+
+### Reference Datum Royalty Flag
+
+If not specified elsewhere in the token's datums, a malicious user could send transactions to a protocol which do not reference the royalty datum. For full assurances, a new optional flag should be added to the reference datum
+
+```
+extra = 
+	{
+		...
+
+		? royalty_included : big_int
```

Comment:
What about using `#6.121[] / #6.122[]` (False / True) instead ?

Reply:
This was my initial idea, and it would certainly work for v1. I have an idea for v2 of this schema to make use of this field to allow for multiple royalty datums to be attached to a policy, with the correct datum to be used for each individual NFT specified in this field.

With that plan in mind, I decided to make the type `big_int` instead, so the type expected by third parties doesn't have to change in future versions. Since computational complexity of Plutus primitives scale with size, performance should be effectively identical.